pkg/tfshim/sdk-v1/instance_diff.go

```diff
@@ -55,6 +55,28 @@ func (d v1InstanceDiff) Attributes() map[string]shim.ResourceAttrDiff {
 	return m
 }
 
+func (d v1InstanceDiff) SetAttribute(key string, attrDiff shim.ResourceAttrDiff) {
+	var t terraform.DiffAttrType
+	switch attrDiff.Type {
+	case shim.DiffAttrInput:
+		t = terraform.DiffAttrInput
+	case shim.DiffAttrOutput:
+		t = terraform.DiffAttrOutput
+	default:
+		t = terraform.DiffAttrUnknown
+	}
+	d.tf.Attributes[key] = &terraform.ResourceAttrDiff{
```

Comment:
Ah this really  is not good, I don't trust that we understand the representation of key in d.tf.Attributes to work correctly for the nested cases. Can we rewrite so we do not need to ever touch things with SetAttribute? I'm my understanding for this change we are not trying to send any new/changed information to TF.